v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/build-system.ts

```diff
@@ -78,7 +78,10 @@ export class SolidityBuildSystemImplementation implements SolidityBuildSystem {
     const localFilesToCompile = (
       await Promise.all(
         this.#options.soliditySourcesPaths.map((dir) =>
-          getAllFilesMatching(dir, (f) => f.endsWith(".sol")),
+          getAllFilesMatching(
+            dir,
+            (f) => f.endsWith(".sol") && !f.endsWith(".t.sol"),
+          ),
```

Comment:
I'm saying here that we should **NOT** compile `.t.sol` files present in solidity sources in the default compile task.

Reply:
looks good